proxy/src/main/java/com/wavefront/agent/preprocessor/PointLineReplaceRegexTransformer.java

```diff
@@ -0,0 +1,48 @@
+package com.wavefront.agent.preprocessor;
+
+import com.google.common.base.Function;
+import com.google.common.base.Preconditions;
+
+import com.yammer.metrics.core.Counter;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.annotation.Nullable;
+
+/**
+ * Replace regex transformer. Performs search and replace on an entire point line string
+ *
+ * Created by Vasily on 9/13/16.
+ */
+public class PointLineReplaceRegexTransformer implements Function<String, String> {
+
+  private final String patternReplace;
+  private final Pattern compiledPattern;
+  @Nullable
+  private final Counter ruleAppliedCounter;
+
+  public PointLineReplaceRegexTransformer(final String patternMatch,
```

Comment:
Probably not here - @Notnull annotation only helps the IDE to give a soft warning if a null value can potentially sneak in (and here it obviously can, as it's being parsed from yaml - hence the validation in the constructor)